### PR TITLE
Fix #154. Don't set nil subs on new blocks

### LINKED ISSUE #154: AsciidoctorDiagram doesn't work with Asciidoctor 1.5.6
Rendering a document containing a diagram with Asciidoctor 1.5.6 and Asciidoctor-Diagram 1.5.4 fails with a Nil error unless substitutions are explicitly defined for that block.
This is because https://github.com/asciidoctor/asciidoctor-diagram/blob/master/lib/asciidoctor-diagram/extensions.rb#L410-L417 returns nil instead of an empty list.

To reproduce try to render this document with Asciidoctor 1.5.6 and Asciidoctor-Diagram 1.5.4: 
```
= Document Title

[ditaa,asciidoctor-diagram-process]
....
                   +-------------+
                   | Asciidoctor |-------+
                   |   diagram   |       |
                   +-------------+       | PNG out
                       ^                 |
                       | ditaa in        |
                       |                 v
 +--------+   +--------+----+    /---------------\
 |        |---+ Asciidoctor +--->|               |
 |  Text  |   +-------------+    |   Beautiful   |
 |Document|   |   !magic!   |    |    Output     |
 |     {d}|   |             |    |               |
 +---+----+   +-------------+    \---------------/
     :                                   ^
     |          Lots of work             |
     +-----------------------------------+
....
```

### PR DESCRIPTION
**Pull request by robertpanzer**
This is an attempt to fix #154 
I have no idea if I really got it right, but this change makes the AsciidoctorJ tests pass with the latest Asciidoctor 1.5.6.

### CODE REVIEW

**Comment by robertpanzer** [comment]
The tests fail on Travis just as they also fail locally for me.
But I am not really able to check why it fails and if these failures are related to the PR.
(In fact I get the same test failures on current master)

**Comment by mojavelinux** [comment]
I believe this is the right change.

When reworking the substitutions behavior in core, I broke the contract of nil as an acceptable value for the subs argument to apply_subs without realizing that Asciidoctor Diagram was using it. Since core never passes nil, the test suite didn't detect the problem.

I've gone ahead and restored support for nil just to be safe, but Asciidoctor Diagram will need to change in the interim in order to work with Asciidoctor 1.5.6. Instead of passing nil subs, pass empty subs as that is what core checks for.

I apologize for messing this up, but there are steps we can take to avoid missing something like this in the future. Core can now trigger dependent builds. If you'd like, I can add asciidoctor-diagram to that list.

File: lib/asciidoctor-diagram/extensions.rb [hunk 1]
```diff
@@ -410,9 +410,8 @@ def checksum
         def resolve_diagram_subs
           if @attributes.key? 'subs'
             subs = @parent_block.resolve_block_subs @attributes['subs'], nil, 'diagram'
```
Comment (mojavelinux) [inline lib/asciidoctor-diagram/extensions.rb]: This assignment is no longer needed.

**Comment by robertpanzer** [comment]
Thanks, @mojavelinux ! 
Removed the assignment.

**Comment by mojavelinux** [comment]
Once this problem is fixed, there is a second problem. It seems like something is getting frozen in extensions that wasn't before. So the which function is failing to assign the resolved command to the class instance. We'll have to address that separately.

**Comment by robertpanzer** [comment]
You mean this?
```
asciidoctor-diagram: ERROR: Failed to generate image: can't modify frozen Asciidoctor::Diagram::PlantUmlBlockProcessor
```
Ditaa probably works because it's a pure Java solution?

**Comment by mojavelinux** [comment]
Correct. Yes, ditaa works because it doesn't have to resolve a command (which is the code that is attempting to modify the instance). There's another case in the mermaid integration. Both are easily solved by using the config map instead, which actually makes the code cleaner anyway.

**Comment by robertpanzer** [comment]
Closing this PR as it's already part of #156

**Pull request closed without merging**